File: backend/EasyPark.Model/Models/ParkingSpot.cs
using System;

namespace EasyPark.Model.Models
{
    public class ParkingSpot
    {
        public int Id { get; set; }
        public int ParkingLocationId { get; set; }
        public string ParkingLocationName { get; set; } = null!;
        public string SpotNumber { get; set; } = null!;
        public string SpotType { get; set; } = null!;
        public bool IsActive { get; set; }
        public bool IsOccupied { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? NextReservationStart { get; set; }
        public DateTime? NextReservationEnd { get; set; }
    }
}

File: frontend/easypark_mobile/lib/models/parking_spot.dart
class ParkingSpot {
  final int id;
  final int parkingLocationId;
  final String parkingLocationName;
  final String spotNumber;
  final String spotType;
  final bool isActive;
  final bool isOccupied;

  ParkingSpot({
    required this.id,
    required this.parkingLocationId,
    required this.parkingLocationName,
    required this.spotNumber,
    required this.spotType,
    required this.isActive,
    required this.isOccupied,
  });

  factory ParkingSpot.fromJson(Map<String, dynamic> json) {
    return ParkingSpot(
      id: json['id'] as int,
      parkingLocationId: json['parkingLocationId'] as int,
      parkingLocationName: json['parkingLocationName'] as String,
      spotNumber: json['spotNumber'] as String,
      spotType: json['spotType'] as String,
      isActive: json['isActive'] as bool,
      isOccupied: json['isOccupied'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'parkingLocationId': parkingLocationId,
      'parkingLocationName': parkingLocationName,
      'spotNumber': spotNumber,
      'spotType': spotType,
      'isActive': isActive,
      'isOccupied': isOccupied,
    };
  }
}
